src/mina-signer/src/nullifier.ts

```diff
@@ -0,0 +1,58 @@
+import { Fp } from '../../bindings/crypto/finite_field.js';
+import { Poseidon } from '../../bindings/crypto/poseidon.js';
+import { Group, PublicKey } from '../../provable/curve-bigint.js';
+import { PrivateKey } from '../../provable/curve-bigint.js';
+import { Field } from '../../provable/field-bigint.js';
+import { Nullifier } from './TSTypes.js';
+
+export { createNullifier };
+/*
+PLUME: An ECDSA Nullifier Scheme for Unique
+Pseudonymity within Zero Knowledge Proofs
+https://eprint.iacr.org/2022/1255.pdf chapter 3 page 14
+*/
+function createNullifier(message: Field, sk: PrivateKey): Nullifier {
```

Comment:
I agree it would improve usability, but I think then we would have to specify the static arrays size in advance, or using some workarounds for the in-snark variant, I made it a one Field argument so both implementations (out-of-snark and in-snark) can be uniform. Or can we pass a dynamic-size array in-snark into Poseidon? :thinking:

Reply:
you can pass any size array into Poseidon, but you're fixing the circuit to always use that size by doing that. I think that's ok